localstack/services/sqs/provider.py

```diff
@@ -808,6 +810,13 @@ def receive_message(
             wait_time_seconds = queue.wait_time_seconds
 
         num = max_number_of_messages or 1
+
+        if num < 1 or num > MAX_NUMBER_OF_MESSAGES:
+            raise InvalidParameterValue(
+                f"Value {num} for parameter MaxNumberOfMessages is invalid. "
+                f"Reason: Must be between 1 and 10, if provided."
+            )
+
         # backdoor to get all messages
         if num == -1:
             num = queue.visible.qsize()
```

Comment:
this will make the control path introduced in #7331 unreachable (it's just below `if num == -1`, so we would have to make an exception for that.